extensions/amp-story/1.0/media-pool.js

```diff
@@ -546,13 +516,14 @@ export class MediaPool {
       new SwapIntoDomTask(placeholderEl)
     ).then(
       () => {
-        this.maybeResetAmpMedia_(ampMediaForPoolEl);
-        this.maybeResetAmpMedia_(ampMediaForDomEl);
         this.enqueueMediaElementTask_(
           poolMediaEl,
           new UpdateSourcesTask(this.win_, sources)
         );
-        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask());
+        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()).then(() => {
```

Comment:
An additional fix here, `maybeResetAmpMedia_` will call `resetOnDomChange` that will call `loadPromise()`. So we don't make this call until the `loadTask` ran, since it will restart the media selection process and `readyState`.

Reply:
Just to be super sure since this being false could introduce really hairy race conditions: Is the expected behavior that calling `loadPromise` either before or after the video has loaded should eventually resolve?  i.e. we don't have to worry about the case where the video has already loaded before this `then` block gets executed?

Reply:
Yep, the `loadPromise` is pretty smart about this!

This was actually the issue, `loadPromise` was detecting that the default sources loaded, not the actual ones, and calling play() too early. Not having default sources fixes the issue, and I added an extra fix to call load() when removing the sources so it resets the `readyState` and all we care about.